### PR TITLE
Integration2 t6 f2 73 rest controller

### PR DESCRIPTION
**Pull request by nala-he**
Finished implementing the following controllers and their interfaces: RestaurantController, FeaturedItemController, UpdateController, and HourController. Passed all tests on postman using our database from Mongodb Atlas.

### CODE REVIEW

File: controllers/RestaurantController.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import RestaurantDao from "../daos/RestaurantDao";
+import RestaurantControllerI from "../interfaces/RestaurantControllerI";
+
+/**
+ * @class RestaurantController Implements RESTful Web service API for restaurants resource
+ * Defines the following HTTP endpoints:
+ * <ul>
+ *     <li>GET /api/restaurants to retrieve all the restaurant instances</li>
+ *     <li>GET /api/restaurants/:rid to retrieve a particular restaurant instance</li>
+ *     <li>GET /api/users/:uid/restaurants to retrieve restaurants for a given user </li>
```
Comment (MaceeQi) [inline controllers/RestaurantController.ts]: Retrieving restaurants for a given user is listed here in jsdoc but not implemented (or vice versa - forgot to delete jsdoc?)

**Review by MaceeQi — COMMENTED** [review]
Tested all the controllers on Postman and everything worked well. One small comment in RestaurantController that needs to be addressed but once that's fixed, I think it's good to merge!

**Review by bovornc — APPROVED** [review]
RestaurantController has a JSDoc that isn't implemented (line 14), but other than that all my postman tests worked fine!

**Comment by nala-he** [comment]
I deleted the jsdoc line 14 that I forgot to delete.

File: controllers/RestaurantController.ts (hunk 1 above)
Comment (nala-he) [reply controllers/RestaurantController.ts]: Yes, I forgot to delete that from jsdoc. Just deleted it.

**Review by MaceeQi — APPROVED** [review]
Thank you for making the changes!

**Pull request merged**